Makefile

```diff
@@ -0,0 +1,20 @@
+CPP     = g++
+CPPOPTS = -g -O3 -Wall -std=c++11 -Iinclude/ -Ideps/ -fprofile-arcs -ftest-coverage
```

Comment:
As an addendum, you'll probably want to do separate builds for the multiple flavors, and I recommend sending them to directories named based on the build flavor.